coalib/settings/Section.py

```diff
@@ -159,6 +160,10 @@ def add_or_create_setting(self,
             val.value = str(val._value) + '\n' + setting._value
             self.append(val, custom_key=key)
         else:
+            if key in self.contents:
+                logging.warning('{} setting has already been defined in '
+                                'section {}. The previous setting will be '
+                                'overridden.'.format(key, self.name))
```

Comment:
shouldn't this be inside `ConfParser.py`? The function docstring itself says

> Adds the value of the setting to an existing setting if there is
> already a setting  with the key. Otherwise creates a new setting.

So this is intended behaviour and we shall not raise a warning here. Instead we should do so during parsing of the file.